crates/bevy_asset/src/lib.rs

```diff
@@ -1110,13 +1116,123 @@ mod tests {
 
             assert_eq!(c_text.text, "c");
             assert_eq!(c_load, LoadState::Loaded);
-            assert_eq!(c_deps, DependencyLoadState::Failed);
-            assert_eq!(c_rec_deps, RecursiveDependencyLoadState::Failed);
+            assert!(matches!(c_deps, DependencyLoadState::Failed(_)));
+            assert!(matches!(
+                c_rec_deps,
+                RecursiveDependencyLoadState::Failed(_)
+            ));
+
+            assert_eq!(asset_server.load_state(a_id), LoadState::Loaded);
+            assert_eq!(
+                asset_server.dependency_load_state(a_id),
+                DependencyLoadState::Loaded
+            );
+            assert!(matches!(
+                asset_server.recursive_dependency_load_state(a_id),
+                RecursiveDependencyLoadState::Failed(_)
+            ));
+
+            assert!(asset_server.is_loaded(a_id));
+            assert!(asset_server.is_loaded_with_dependencies(a_id));
+            assert!(!asset_server.is_loaded_with_recursive_dependencies(a_id));
 
             Some(())
         });
     }
 
+//     #[test]
```

Comment:
Remember to uncomment this :)

Reply:
This test runs and passes on my machine but is actually breaking the build and causing multiple tests to fail in CI for some reason. Still trying to isolate the cause.